prov/efa/src/rdm/efa_rdm_msg.c

```diff
@@ -978,6 +966,7 @@ ssize_t efa_rdm_msg_recv(struct fid_ep *ep_fid, void *buf, size_t len,
 	struct efa_rdm_ep *ep;
 
 	ep = container_of(ep_fid, struct efa_rdm_ep, base_ep.util_ep.ep_fid.fid);
+	assert(len <= ep->max_msg_size);
```

Comment:
I don't think we need to apply max_msg_size for recv. Though the documentation doesn't say that, I think it's totally fine that user post a bigger recv buffer than the incoming msg size, i.e. the max_msg_size only applies to transmission.

Reply:
Likewise for RMA read?

Reply:
RMA read is still a transmission, so the limit applies.